### PR TITLE
Add mission deserializer

### PR DESCRIPTION
**Pull request by yas1234**
Add MissionDeserializer
Add tests for MissionDeserializer
Add new message type for Fly Zones
Remove ServerInfoDeserializer

### CODE REVIEW

**Comment by anassinator** [comment]
I don't think you pushed everything. The builds are failing because your new messages aren't there.

**Comment by yas1234** [comment]
I forgot to add the FlyZone message files to the last commit, should work now

**Comment by anassinator** [comment]
The tests are still failing because you removed server info

**Comment by anassinator** [comment]
You can run the tests with `catkin_make run_tests`

**Comment by yas1234** [comment]
Yeah, I will remove the server info test in the client tests

**Comment by yas1234** [comment]
Done

**Comment by anassinator** [comment]
Ok, I'll take a closer look tomorrow, but everything seems good at a glance.
Just make sure to PEP8-ify everything. So fix the missing newlines between functions, periods at end of comments, spacing, formatting and whatnot. Also add comments to the new message types for an idea of the units being used.

File: msg/FlyZone.msg [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Header header
+
+float64 max_alt #Maximum altitude in meters.
+float64 min_alt #Minimum altitude in meters.
+geometry_msgs/Polygon polygon #Polygon with points in utm coordinates.
```
Comment (anassinator) [inline msg/FlyZone.msg]: stamp polygon message for rviz support
remove header
rename `polygon` to `zone`

File: src/interop/serializers.py [hunk 2]
```diff
@@ -66,28 +68,224 @@ def iso8601_to_rostime(iso):
...
+                                                      waypoint["longitude"])
+                point.x = easting
+                point.y = northing
+                flyzone.polygon.points.append(point)
+            flyzones.flyzones.append(flyzone)
+
+        return flyzones
+
+    @staticmethod
+    def __get_waypoints(json,frame):
```
Comment (anassinator) [inline src/interop/serializers.py]: json, frame

File: src/interop/serializers.py [hunk 3]
```diff
@@ -66,28 +68,224 @@ def iso8601_to_rostime(iso):
...
+
+    @staticmethod
+    def __emergent_object(json,frame):
+        """
+        Deserializes the last known location of the emergent target to a 
+        ros message of type PointStamped.
+
+        Args:
+            json: JSON dict with the corresponding location.
+            frame: frame for the point.
```
Comment (anassinator) [inline src/interop/serializers.py]: Capitalize docstring
Newline between end of args and beginning of returns

File: src/interop/serializers.py [hunk 4]
```diff
@@ -66,28 +68,224 @@ def iso8601_to_rostime(iso):
...
+class MissionDeserializer(object):
 
-    """Server information deserializer."""
+    """Mission information deserializer."""
 
-    @classmethod
-    def from_json(cls, json):
-        """Deserializes server information into a tuple of standard ROS
-        messages.
+    @staticmethod
```
Comment (anassinator) [inline src/interop/serializers.py]: classmethod

File: test/test_serializers.py [hunk 5]
```diff
@@ -19,25 +19,145 @@
...
-        # Compare properties
-        self.assertEqual(server_info[0].data, json["message"])
-        msg_time = serializers.iso8601_to_rostime(json["message_timestamp"])
-        self.assertEqual(server_info[1], msg_time)
-        server_time = serializers.iso8601_to_rostime(json["server_time"])
-        self.assertEqual(server_info[2], server_time)
+    def test_mission_deserializer(self):
+        "Tests the mission deserializer."
+        json =   {
+		    "id": 1,
```
Comment (anassinator) [inline test/test_serializers.py]: tab inwards

**Comment by yas1234** [comment]
Fixed Flyzone and style errors

**Pull request merged**